File: bitrepository-integrity-service/src/main/java/org/bitrepository/integrityservice/utils/FileSizeUtils.java
package org.bitrepository.integrityservice.utils;

public class FileSizeUtils {

    private static final int unitSize = 1024;
    private static final long kiloSize = unitSize;
    private static final long megaSize = kiloSize * unitSize;
    private static final long gigaSize = megaSize * unitSize;
    private static final long teraSize = gigaSize * unitSize;
    private static final long petaSize = teraSize * unitSize;
    private static final long zetaSize = petaSize * unitSize;
    
    private static final String bytePostfix = "B";
    private static final String kiloPostfix = "KB";
    private static final String megaPostfix = "MB";
    private static final String gigaPostfix = "GB";
    private static final String teraPostfix = "TB";
    private static final String petaPostfix = "PB";
    private static final String zetaPostfix = "ZB";
    
    
    public static String toHumanShort(long size) {
        if(size >= zetaSize) {
            return formatShortZeta(size);
        } else if(size >= petaSize) {
            return formatShortPeta(size);
        } else if(size >= teraSize) {
            return formatShortTera(size);
        } else if(size >= gigaSize) {
            return formatShortGiga(size);
        } else if(size >= megaSize) {
            return formatShortMega(size);
        } else if(size >= kiloSize) {
            return formatShortKilo(size);
        } else {
            return formatShortByte(size);
        }
    }
    
    private static String formatShortZeta(long size) {
        int wholeZB = (int) (size / zetaSize);
        return wholeZB + zetaPostfix;
    }
    
    private static String formatShortPeta(long size) {
        int wholePB = (int) (size / petaSize);
        return wholePB + petaPostfix;
    }
    
    private static String formatShortTera(long size) {
        int wholeTB = (int) (size / teraSize);
        return wholeTB + teraPostfix;
    }
    
    private static String formatShortGiga(long size) {
        int wholeGB = (int) (size / gigaSize);
        return wholeGB + gigaPostfix;
    }
    
    private static String formatShortMega(long size) {
        int wholeMB = (int) (size / megaSize);
        return wholeMB + megaPostfix;
    }
    
    private static String formatShortKilo(long size) {
        int wholeKB = (int) (size / kiloSize);
        return wholeKB + kiloPostfix;
    }
    
    private static String formatShortByte(long size) {
        return size + bytePostfix;
    }
}

File: bitrepository-webclient/src/main/webapp/FileSizeUtils.js

  function FileSizeUtils() {
    
    var unitSize = 1024;
    var byteSize = 1;
    var kiloSize = byteSize * unitSize;
    var megaSize = kiloSize * unitSize;
    var gigaSize = megaSize * unitSize;
    var teraSize = gigaSize * unitSize;
    var petaSize = teraSize * unitSize;
    var exaSize = petaSize * unitSize;

    var bytePostFix = "B";
    var kiloPostFix = "KB";
    var megaPostFix = "MB";
    var gigaPostFix = "GB";
    var teraPostFix = "TB";
    var petaPostFix = "PB";
    var exaPostfix = "EB";


    this.toHumanUnit = function(size) {
      if (size == null) {
        return bytePostfix;
      }
      if (size >= exaSize) {
        return exaPostfix;
      } else if (size >= petaSize) {
        return petaPostfix;
      } else if (size >= teraSize) {
        return teraPostfix;
      } else if (size >= gigaSize) {
        return gigaPostfix;
      } else if (size >= megaSize) {
        return megaPostfix;
      } else if (size >= kiloSize) {
        return kiloPostfix;
      } else {
        return bytePostfix;
      }
    }

    this.getByteSize = function(unit) {
      if (exaPostfix.equals(unit)) {
        return exaSize;
      } else if (petaPostfix.equals(unit)) {
        return petaSize;
      } else if (teraPostfix.equals(unit)) {
        return teraSize;
      } else if (petaPostfix.equals(unit)) {
        return petaSize;
      } else if (gigaPostfix.equals(unit)) {
        return gigaSize;
      } else if (megaPostfix.equals(unit)) {
        return megaSize;
      } else if (kiloPostfix.equals(unit)) {
        return kiloSize;
      } else if (bytePostfix.equals(unit)) {
        return byteSize;
      } else {
        return byteSize;
      }
    }
  }
